clang/include/clang/Basic/Attr.td

```diff
@@ -1197,6 +1197,13 @@ def SYCLDevice : InheritableAttr {
   let Documentation = [SYCLDeviceDocs];
 }
 
+def SYCLGlobalVar : InheritableAttr {
+  let Spellings = [GNU<"sycl_global_var">];
+  let Subjects = SubjectList<[GlobalVar]>;
+  let LangOpts = [SYCLIsDevice];
```

Comment:
So when written in a host compilation, we *want* an unknown attribute warning? That seems suspicious.

Reply:
Variables marked with this attribute are intended to be used from sycl device-side code, and the attribute or whole variable is expected to be wrapped in a \_\_SYCL_DEVICE_ONLY\_\_. I went with the more conservative LangOpts choice that seemed to satisfy that need, but I can broaden that if we think it's useful to do so.

Reply:
I have the sneaking suspicion that there is a delta between "expected to be wrapped" and "will always be wrapped". :-D We can keep the more restrictive options now and relax them later if need be.